### PR TITLE
doc: Restructure installation guides

### PR DESCRIPTION
**Pull request by tgraf**
* Massively simplify the standard installation guide by moving out optional
  sections
* Add reference to kubespray installer

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5259)

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

File: Documentation/kubernetes/install/kubespray.rst [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    Please use the official rendered version released here:
+    http://docs.cilium.io
+
+.. _k8s_install_kubespray:
+
+****************************
+Installation using Kubespray
+****************************
+
+Cilium is supported in the kubespray installer. Just following the standard guide:
```
Comment (ianvernon) [inline Documentation/kubernetes/install/kubespray.rst]: following --> follow

**Review by ianvernon — APPROVED** [review]
LGTM, one minor grammatical nitpick.

**Comment by tgraf** [comment]
test-docs-please

**Pull request merged**